src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
```

Comment:
PrimathonAICodeReviewer

Consider adding null checks for `request` and `request.headers['x-feedback-secret']` to avoid potential runtime errors.

```diff
+     const request = context.switchToHttp().getRequest();
+     const secret = Env.SECRET.FEEDBACK;
+     const requestSecret = request?.headers['x-feedback-secret'];
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->